app/models/grouping.rb

```diff
@@ -513,12 +513,16 @@ def number_of_submitted_files
 
   # Returns last modified date of the assignment_folder in this grouping's repository
   def assignment_folder_last_modified_date
-    repo = self.group.repo
-    rev = repo.get_latest_revision
-    last_date = rev.timestamp
-
-    repo.close()
-    last_date
+    revision = nil
+    group.access_repo do |repo|
+      # we use this function because it allows to specify a path
+      revision = repo.get_revision_by_timestamp(Time.zone.now, assignment.repository_folder)
+      # an alternative could have been:
+      # repo.get_latest_revision.directories_at_path('')[assignment.repository_folder].last_modified_date
+      # but in git, the latter returns commit times instead of push times, and would be less efficient because it has to
+      # walk through the entire history
+    end
+    revision.timestamp
```

Comment:
I'm wondering about a possible `nil` error here. I guess in the old version, `rev` was guaranteed not to be nil, but I'm not sure this is true here.

Reply:
That function always returns a revision for git and svn, with the first revision as a fallback.
(I'm sure about git, and the svn comments say that at least)